cognite/client/data_classes/extractionpipelines.py

```diff
@@ -301,7 +301,7 @@ def __init__(
         self.revision = revision
         self.description = description
         self.created_time = created_time
-        self.cognite_client = cognite_client
+        self._cognite_client = cognite_client
```

Comment:
I see that we do some magic in `__new__`, but this only sets `self._cognite_client` if `cognite_client` was given, it does not pop it from kwargs, and as such, subclasses are free to set it as non-private 🤔